src/dao/WavesDAO.js

```diff
@@ -0,0 +1,253 @@
+            time: tx.time,
+            from: tx.from,
+            to: tx.to,
+            symbol: this._symbol,
+            value: new Amount(tx.value, this._symbol),
+            fee: new Amount(tx.fee, this._symbol),
+            blockchain: BLOCKCHAIN_WAVES,
+          }))
+        }
+      }
+    } catch (e) {
+      // eslint-disable-next-line
+      console.warn('WavesDAO getTransfer', e)
+    }
+    return txs
+  }
+
+  watch (/*account*/): Promise {
+    return Promise.all([
+      this.watchTransfer(),
+      this.watchBalance(),
+    ])
+  }
+
+  async watchTransfer () {
+    this._wavesProvider.addListener(EVENT_TX, async ({ tx }) => {
+      if (tx.unconfirmed) {
+        if (!this._mosaic) {
+          if (!tx.mosaics) {
+            this.emit(EVENT_NEW_TRANSFER, this._createXemTxModel(tx))
+          }
+        } else {
+          if (tx.mosaics && (this._namespace in tx.mosaics)) {
+            this.emit(EVENT_NEW_TRANSFER, this._createMosaicTxModel(tx))
+          }
+        }
+      }
+    })
+  }
+
+  _createWavesTxModel (tx) {
+    return new TxModel({
+      txHash: tx.txHash,
+      // blockHash: tx.blockhash,
+      // blockNumber: tx.blockheight,
+      blockNumber: null,
+      time: tx.time,
+      from: tx.from || tx.signer,
+      to: tx.to,
+      value: new Amount(tx.value, this._symbol),
+      fee: new Amount(tx.fee, this._symbol),
+      blockchain: BLOCKCHAIN_WAVES,
+    })
+  }
+
+  _createAssetTxModel (tx) {
+    return new TxModel({
+      txHash: tx.txHash,
+      // blockHash: tx.blockhash,
+      // blockNumber: tx.blockheight,
+      blockNumber: null,
+      time: tx.time,
+      from: tx.from || tx.signer,
+      to: tx.to,
+      value: new Amount(tx.assets[this._assetId], this._symbol),
+      fee: new Amount(tx.fee, WAVES_WAVES_SYMBOL),
+      blockchain: BLOCKCHAIN_WAVES,
+    })
+  }
+
+  async watchBalance () {
+    this._wavesProvider.addListener(EVENT_BALANCE, async ({ account, time, balance }) => {
+      this.emit(EVENT_UPDATE_BALANCE, {
+        account,
+        time,
+        balance: readBalanceValue(this._symbol, balance, this._asset),
+      })
+    })
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  async watchApproval (callback) {
+    // Ignore
+  }
+
+  async stopWatching () {
+    // Ignore
+  }
+
+  resetFilterCache () {
+    // do nothing
+  }
+
+  async fetchToken () {
+    if (!this.isInitialized()) {
+      const message = `${this._symbol} support is not available`
+      // eslint-disable-next-line
+      console.warn(message)
+      throw new Error(message)
+    }
+
+    const token = new TokenModel({
+      name: this._name,
+      decimals: this._decimals,
+      symbol: this._symbol,
+      isOptional: false,
+      isFetched: true,
+      blockchain: BLOCKCHAIN_WAVES,
+    })
+
+    return token
+  }
+}
+
+//TODO WHY WE NEED SYMBOL AND ASSET DESCRIPTION IF SYMBOL IS ENOUGH
+function readBalanceValue (symbol, balance, asset = null) {
```

Comment:
Mosaic description required to sign NEM Mosaic transfers + Mosaic definition contains info about decimals places in Mosaic amounts.
Do not sure if we need full asset description for Waves.